.gitlab-ci.yml

```diff
@@ -0,0 +1,347 @@
+stages:
+  - base
+  - build
+  - package
+  - upload
+  - release
+
+variables:
+  GITIAN_DIR: /opt/gitian-builder
+  PACKAGE_REGISTRY_URL: "${CI_API_V4_URL}/projects/${CI_PROJECT_ID}/packages/generic/peercoin/${CI_COMMIT_TAG}"
+
+workflow:
+  rules:
+    - if: $CI_MERGE_REQUEST_ID
+      when: never
+    - when: always
+
+.build docker job:
+  stage: base
+  image: docker:stable
+  variables:
+    DOCKER_BUILDKIT: 1
+    GIT_STRATEGY: none
+  rules:
+    - if: $CI_PIPELINE_SOURCE != "trigger"
+  services:
+    - docker:stable-dind
+  before_script:
+    - echo $CI_JOB_TOKEN | docker login -u gitlab-ci-token --password-stdin $CI_REGISTRY
+  after_script:
+    - docker tag $IMAGE $IMAGE:$CI_COMMIT_REF_SLUG
+    - docker push $IMAGE:$CI_COMMIT_REF_SLUG
+    - if [ "$CI_COMMIT_REF_NAME" == "$CI_DEFAULT_BRANCH" ]; then docker push $IMAGE; fi;
+
+build gitian builder:
+  extends: .build docker job
+  variables:
+    IMAGE: $CI_REGISTRY_IMAGE/gitian/builder
+  script:
+    - |
+      docker build --tag $IMAGE --cache-from $IMAGE:$CI_COMMIT_REF_SLUG --build-arg BUILDKIT_INLINE_CACHE=1 -<<EOF
+        FROM docker:stable-dind
+        RUN apk add curl make ruby wget git sed
+        RUN git clone https://github.com/devrandom/gitian-builder.git ${GITIAN_DIR}
+        RUN mkdir ${GITIAN_DIR}/inputs ${GITIAN_DIR}/var
+        RUN wget -N https://github.com/phracker/MacOSX-SDKs/releases/download/11.3/MacOSX10.14.sdk.tar.xz \
```

Comment:
This might be debatable and should be adjusted when needed.

Reply:
Can you please explain what is being done here, exactly, and why it's debatable?

Reply:
Supposedly Apple is very stingy about using their SDKs in a pre-packaged fashion. I'm no lawyer so I do not know how this would affect any projects using this download location.  
In addition, for this example, I do not check the checksum of the downloaded file.

Reply:
It's fine if you use such pre-packaged SDK.